arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Either.kt

```diff
@@ -897,8 +897,14 @@ public sealed class Either<out A, out B> {
    * <!--- KNIT example-either-36.kt -->
    * <!--- TEST lines.isEmpty() -->
    */
-  public inline fun <C> map(f: (right: B) -> C): Either<A, C> =
-    flatMap { Right(f(it)) }
+  @OptIn(ExperimentalContracts::class)
+  public inline fun <C> map(f: (right: B) -> C): Either<A, C> {
+    contract {
+      callsInPlace(f, InvocationKind.EXACTLY_ONCE)
```

Comment:
All of these methods are `AT_MOST_ONCE` since if you call `map` on `Either.Left` then `f` won't be executed, same for `mapLeft` on `Either.Right` and similarly for `onLeft` and `onRight`.

Reply:
Oh right I understand, I'll correct it